src/oca_github_bot/manifest.py

```diff
@@ -199,3 +199,26 @@ def get_odoo_series_from_version(version):
     if not series:
         raise OdooSeriesNotDetected()
     return tuple(int(s) for s in series.split("."))
+
+
+def _user_can_push(gh, org, repo, username, addons_dir, target_branch):
+    """
+    Check if a user is allowed to push.
+    addons_dir must be a git clone of the PR's branch.
```

Comment:
Check if the user is maintainer of all modified addons.

Assuming, ``addons_dir`` is a git clone of an addons repository, return true if ``username`` is declared in the ``maintainers`` key on the target branch, for all addons modified in the current branch compared to the ``target_branch``.